csharp/src/Ice/IceSSL/TransceiverI.cs

```diff
@@ -393,34 +406,97 @@ private bool startAuthenticate(IceInternal.AsyncCallback callback, object state)
             {
                 if(!_incoming)
                 {
-                    //
                     // Client authentication.
-                    //
-                    _writeResult = _sslStream.AuthenticateAsClientAsync(_host,
-                                                                        _instance.certs(),
-                                                                        _instance.protocols(),
-                                                                        _instance.checkCRL() > 0);
-                    _writeResult.ContinueWith(task => callback(state), TaskScheduler.Default);
+                    if (_instance.initializationData().clientAuthenticationOptions
+                        is SslClientAuthenticationOptions clientAuthenticationOptions)
+                    {
+                        _writeResult = _sslStream.AuthenticateAsClientAsync(clientAuthenticationOptions);
+                        _writeResult.ContinueWith(
+                            task =>
+                            {
+                                try
+                                {
+                                    _verified = task.IsCompletedSuccessfully;
```

Comment:
AuthenticateAsClientAsync throws `AuthenticationException` when authentication fails.

Reply:
I would add a comment.

Reply:
Do you plan to clean-up all this code in a follow-up PR with async/await?